test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -220,6 +227,7 @@
     private Function<Client, Client> clientWrapper;
 
     public InternalTestCluster(String nodeMode, long clusterSeed, Path baseDir,
+                               int numMasterNodes,
```

Comment:
I see what you are saying, though maybe people want to make sure there is only one master for simplicity - I think it's OK to keep this the same as numClientNodes for example? I can change both to a boolean ( maybe) but that should be a different PR imo.